packages/cw-orch-core/src/contract/deploy.rs

```diff
@@ -90,7 +91,12 @@ pub trait Deploy<Chain: CwEnv>: Sized {
     ) -> anyhow::Result<HashMap<String, Self>> {
         let hash_networks: HashMap<String, (Chain, Self::DeployData)> = networks
             .iter()
-            .map(|(c, d)| Ok::<_, Chain::Error>((c.block_info()?.chain_id, (c.clone(), d.clone()))))
+            .map(|(c, d)| {
+                Ok::<_, <Chain as QueryHandler>::Error>((
+                    QueryHandler::block_info(c)?.chain_id,
```

Comment:
There is only one block_info function, why indicate that it's the QueryHandler's one ? 

```suggestion
                    c.block_info()?.chain_id,
```